src/lib.rs

```diff
@@ -73,17 +73,18 @@ pub use signature::SignedRequest;
 pub use request::default_tls_client;
 pub use region::default_region;
 
-mod param;
-mod region;
-mod request;
-mod xmlerror;
-mod xmlutil;
-mod serialization;
+pub mod param;
```

Comment:
An option here is to put `mock` in its own crate that each service imports uses in `dev-dependencies`. It then never needs to be exposed, but could be published separately if users want it (though if #620 gets merged, I don't think it would ever need to be exposed).

I've been [playing around](https://github.com/mthjones/rusoto/blob/split-everything/mock/Cargo.toml) with it a bit, seems pretty easy to set up. Unless it causes a circular dependency... I split the core functionality out from `rusoto`, so I'm not sure it would be possible without that.